### ISSUE TITLE
[FEATURE] Ensure info metrics will have compatibility with prometheus OTEL implementation

### ISSUE THREAD

**Issue by kgeckhart** [nerdswords/yet-another-cloudwatch-exporter#1436]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

YACE exports info for resource metadata like tags. This normalizes the data off of the CloudWatch metrics and produces a simplified metric model that can be built up via prometheus joins. 

The problem is that YACE chose to export these info metrics with a value of 0. This goes against the accepted prometheus implementation for info metrics where the value exported is 1, https://www.robustperception.io/why-info-style-metrics-have-a-value-of-1/. So far this hasn't really been a big deal other than joins need to use a `+` operator vs the a `*` to avoid breaking values.

But now that prometheus is moving to be more otel compatible they are planning to implement simplified query support for info metrics, https://github.com/prometheus/prometheus/issues/12629. The problem is that whatever is done with prometheus is likely to be incompatible with YACE as YACE does not follow the accepted standard.

We should find a way to safely migrate from exporting a 0 value from info metrics to a 1 to ensure we reap the benefits of the upcoming info metric work.

**Comment by Pokom** [comment]
I actually just ran into this trying to join data from https://github.com/grafana/cloudcost-exporter/blob/main/docs/metrics/aws/ec2.md to calculate the costs of an instance and was surprised to see the value is returned as zero. Is there any specific reason _why_ info metrics are exported as zero? IE will it be a backwards incompatible change?

**Comment by kgeckhart** [comment]
> Is there any specific reason _why_ info metrics are exported as zero? IE will it be a backwards incompatible change?

I'm not sure as I think this one largely predates me and @cristiangreco. I found that it's been 0 for at least 6 years now, https://github.com/nerdswords/yet-another-cloudwatch-exporter/pull/4/files#diff-dec443ff93e42df03c5316555338fccf81a2cb1ec4a646276a02cf034e6895cdR88-R127. Pretty sure it's been zero since the info metrics existed.

Changing it to 1 now is going to break anyone who currently has queries which depend on the value being zero 😬

**Comment by jeschkies** [comment]
Would we have to add `deployment_environment`?

**Comment by kgeckhart** [comment]
> Would we have to add `deployment_environment`?

Yes, I'll take a pass at the ticket now that the info function implementation is better understood. ATM the value being 0 isn't a problem but the structure of the current info metrics is. The function expects an `instance` label + `job` label where we currently use `name` + `job`. We should also look to move to `target_info` vs `aws_<service>_info` which will work very cleanly with `info` vs needing more complex joins still.
